src/client/connection.ts

```diff
@@ -43,13 +44,28 @@ export function getEventRouter(): EventRouter<Events> {
 /**
  * Promise to the channel object that allows us to connect to the client
  */
-let channelPromise: Promise<ChannelClient>|null = null;
+let channelPromise: Promise<ChannelClient> | null = null;
+const hasDOMContentLoaded = new DeferredPromise<void>();
 
 if (typeof fin !== 'undefined') {
-    getServicePromise();
+    if (getServiceIdentity().name !== fin.Window.me.name && getServiceIdentity().uuid !== fin.Window.me.uuid) {
```

Comment:
Is this safe? We don't know we have a fixed service identity until after `getServicePromise` is called

Reply:
Removed the checks.